job/server/src/main/java/alluxio/job/transform/format/TableReader.java

```diff
@@ -23,30 +23,20 @@
  */
 public interface TableReader extends Closeable {
   /**
-   * @param scheme the scheme of the path, e.g. "alluxio", "file"
-   * @param path the path to the file representing the table
-   * @return the reader for the table
+   * @param uri the URI to the input
+   * @return the reader for the input
    * @throws IOException when failed to create the reader
    */
-  static TableReader create(String scheme, String path) throws IOException {
-    if (Format.isCsv(path)) {
-      return CsvReader.create(scheme, path);
-    } else if (Format.isParquet(path)) {
-      return ParquetReader.create(scheme, path);
+  static TableReader create(AlluxioURI uri) throws IOException {
+    if (Format.isCsv(uri.getPath())) {
+      return CsvReader.create(uri);
+    } else if (Format.isParquet(uri.getPath())) {
```

Comment:
We can determine a file to be of parquet format by reading some magic header of the file. But for production systems, normally they should end up with `.parquet` or `.csv` or `.csv.gz`.

Reply:
oh that explains it.. this is precisely why I don't see any files generated... because it detected no files.. 
I think this current approach is problematic.. Hive creates random files such as `_SUCCESS` in the partition directories, those files will likely cause an exception to be thrown..
Determining file types from the suffix is just not reliable.. There is information about file format in the Partition information though.  It is in the `StorageDescriptor`

```# Storage Information         | NULL                                               | NULL                        |
| SerDe Library:                | org.apache.hadoop.hive.ql.io.parquet.serde.ParquetHiveSerDe | NULL                        |
| InputFormat:                  | org.apache.hadoop.hive.ql.io.parquet.MapredParquetInputFormat | NULL                        |
| OutputFormat:                 | org.apache.hadoop.hive.ql.io.parquet.MapredParquetOutputFormat | NULL                        |```

Reply:
@chcn FYI, `https://cwiki.apache.org/confluence/display/Hive/SerDe`. We probably would want to open a separate PR for the determining file format commit.

Reply:
will try to resolve this in another PR

Reply:
I still think production systems should have valid suffixes for the real data files